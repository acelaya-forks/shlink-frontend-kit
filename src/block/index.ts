export * from './Message';
export * from './Result';
export * from './SimpleCard';
