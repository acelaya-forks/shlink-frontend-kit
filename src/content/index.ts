export * from './Details';
export * from './Listbox';
export * from './Table';
