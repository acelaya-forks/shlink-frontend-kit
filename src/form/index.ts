export * from './Checkbox';
export * from './ToggleSwitch';
export * from './InputFormGroup';
export * from './LabeledFormGroup';
export * from './SearchField';
