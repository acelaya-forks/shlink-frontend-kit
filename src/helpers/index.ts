export * from './numbers';
export * from './pagination';
