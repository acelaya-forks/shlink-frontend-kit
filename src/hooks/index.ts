import { useRef, useState } from 'react';
import { v4 as uuid } from 'uuid';

type ToggleResult = [boolean, () => void, () => void, () => void];

export const useToggle = (initialValue = false): ToggleResult => {
  const [flag, setFlag] = useState<boolean>(initialValue);
  return [flag, () => setFlag(!flag), () => setFlag(true), () => setFlag(false)];
};

export const useDomId = (): string => {
  const { current: id } = useRef(`dom-${uuid()}`);
  return id;
};

export const useElementRef = <T>() => useRef<T | null>(null);
