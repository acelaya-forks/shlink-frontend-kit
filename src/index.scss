@import './utils/ResponsiveTable';
@import './theme/theme';
