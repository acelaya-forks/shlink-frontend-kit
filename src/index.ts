export * from './block';
export * from './form';
export * from './hooks';
export * from './navigation';
export * from './ordering';
