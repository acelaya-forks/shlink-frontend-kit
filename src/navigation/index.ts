export * from './Dropdown';
export * from './LinkButton';
export * from './Menu';
export * from './NavBar';
export * from './NavPills';
export * from './Paginator';
export * from './RowDropdown';
