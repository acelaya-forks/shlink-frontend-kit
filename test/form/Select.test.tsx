import { render } from '@testing-library/react';
import type { SelectProps } from '../../src';
import { Select } from '../../src';
import { checkAccessibility } from '../__helpers__/accessibility';

describe('<Select />', () => {
  const setUp = (props: Omit<SelectProps, 'children'> = {}) => render(
    <Select {...props}>
      <option>Foo</option>
      <option>Bar</option>
    </Select>,
  );

  it('passes a11y checks', () => checkAccessibility(setUp({ 'aria-label': 'Select option' })));

  it.each([
    {},
    { size: 'sm' as const },
    { size: 'md' as const },
    { size: 'lg' as const },
    { feedback: 'error' as const },
    { disabled: true },
  ])('renders as expected based on provided props', (props) => {
    const { container } = setUp(props);
    expect(container).toMatchSnapshot();
  });
});
