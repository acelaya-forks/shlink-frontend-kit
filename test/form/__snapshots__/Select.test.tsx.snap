// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Select /> > renders as expected based on provided props 1`] = `
<div>
  <select
    class="tw:w-full tw:appearance-none tw:pr-9 tw:bg-(image:--chevron-down) tw:bg-no-repeat tw:focus-ring tw:rounded-md tw:border tw:border-lm-input-border tw:dark:border-dm-input-border tw:pl-3 tw:py-1.5 tw:bg-lm-primary tw:dark:bg-dm-primary tw:group-[&]/card:bg-lm-input tw:group-[&]/card:dark:bg-dm-input"
    style="background: right 0.75rem center / 16px 12px;"
  >
    <option>
      Foo
    </option>
    <option>
      Bar
    </option>
  </select>
</div>
`;

exports[`<Select /> > renders as expected based on provided props 2`] = `
<div>
  <select
    class="tw:w-full tw:appearance-none tw:pr-9 tw:bg-(image:--chevron-down) tw:bg-no-repeat tw:focus-ring tw:rounded-md tw:border tw:border-lm-input-border tw:dark:border-dm-input-border tw:pl-2 tw:py-1 tw:text-sm tw:bg-lm-primary tw:dark:bg-dm-primary tw:group-[&]/card:bg-lm-input tw:group-[&]/card:dark:bg-dm-input"
    style="background: right 0.75rem center / 16px 12px;"
  >
    <option>
      Foo
    </option>
    <option>
      Bar
    </option>
  </select>
</div>
`;

exports[`<Select /> > renders as expected based on provided props 3`] = `
<div>
  <select
    class="tw:w-full tw:appearance-none tw:pr-9 tw:bg-(image:--chevron-down) tw:bg-no-repeat tw:focus-ring tw:rounded-md tw:border tw:border-lm-input-border tw:dark:border-dm-input-border tw:pl-3 tw:py-1.5 tw:bg-lm-primary tw:dark:bg-dm-primary tw:group-[&]/card:bg-lm-input tw:group-[&]/card:dark:bg-dm-input"
    style="background: right 0.75rem center / 16px 12px;"
  >
    <option>
      Foo
    </option>
    <option>
      Bar
    </option>
  </select>
</div>
`;

exports[`<Select /> > renders as expected based on provided props 4`] = `
<div>
  <select
    class="tw:w-full tw:appearance-none tw:pr-9 tw:bg-(image:--chevron-down) tw:bg-no-repeat tw:focus-ring tw:rounded-md tw:border tw:border-lm-input-border tw:dark:border-dm-input-border tw:pl-4 tw:py-2 tw:text-xl tw:bg-lm-primary tw:dark:bg-dm-primary tw:group-[&]/card:bg-lm-input tw:group-[&]/card:dark:bg-dm-input"
    style="background: right 0.75rem center / 16px 12px;"
  >
    <option>
      Foo
    </option>
    <option>
      Bar
    </option>
  </select>
</div>
`;

exports[`<Select /> > renders as expected based on provided props 5`] = `
<div>
  <select
    class="tw:w-full tw:appearance-none tw:pr-9 tw:bg-(image:--chevron-down) tw:bg-no-repeat tw:focus-ring-danger tw:rounded-md tw:border tw:border-danger tw:pl-3 tw:py-1.5 tw:bg-lm-primary tw:dark:bg-dm-primary tw:group-[&]/card:bg-lm-input tw:group-[&]/card:dark:bg-dm-input"
    style="background: right 0.75rem center / 16px 12px;"
  >
    <option>
      Foo
    </option>
    <option>
      Bar
    </option>
  </select>
</div>
`;

exports[`<Select /> > renders as expected based on provided props 6`] = `
<div>
  <select
    class="tw:w-full tw:appearance-none tw:pr-9 tw:bg-(image:--chevron-down) tw:bg-no-repeat tw:focus-ring tw:rounded-md tw:border tw:border-lm-input-border tw:dark:border-dm-input-border tw:pl-3 tw:py-1.5 tw:bg-lm-disabled-input tw:dark:bg-dm-disabled-input"
    disabled=""
    style="background: right 0.75rem center / 16px 12px;"
  >
    <option>
      Foo
    </option>
    <option>
      Bar
    </option>
  </select>
</div>
`;
